### DISCUSSION TITLE
Need help testing alternative benchmarking tool output

### DISCUSSION THREAD

**Discussion by sonicrules1234** [PlummersSoftwareLLC/PrimeView#39] [Show and tell]
I'm looking to test it to see if it's compat output can be parsed properly by PrimeView.  Can somebody see if the output can be parsed by primeview for me?

https://github.com/sonicrules1234/plummerprimes 

Can't figure out how to attach the json file.  I'll link to a gist of it instead.
https://gist.github.com/sonicrules1234/d3c4dd9461d9025dc5403400ebf304b8

**Comment by rbergen** [comment]
I've had a look, and I see a couple of issues with your output:

- the names of the properties in the `machine.cpu.cache` object (`l1d`, `l1i`, ...) should be lowercase
- the value of the `implementation` property in the `results` objects must be lowercase

**Reply by sonicrules1234** [reply]
Thanks, I'll fix that.

**Comment by sonicrules1234** [comment]
How does https://gist.github.com/ab2e3b8f86577dd341088ef060064d7b look?

**Reply by rbergen** [reply]
It looks better. :)
I'll pull it through a local build of PrimeView when I have the time to set it up, to do a real-life test. It could be a few days before I get around to that.

**Reply by sonicrules1234** [reply]
Thanks, it is now updated.

**Reply by sonicrules1234** [reply]
Were you able to get a chance to do this?

**Reply by rbergen** [reply]
Not yet. I have some time off next week, and hope to find the time then.

**Reply by rbergen** [reply]
@sonicrules1234 Ok, so I've loaded up the report in a local build of PrimeView. From a JSON structure perspective, it seems to work fine. I did notice that there are quite a few duplicate results in the JSON file in question, like some ManDeJan&ityonemo&SpexGuy Zig implementations, and some chrvanorle Java implementations. They seem to vary only in the contents of the `tags` object, and even those are sometimes duplicated verbatim. You may have done this to enrich the result data set to test with, but if you didn't this is certainly something to look at.

**Reply by sonicrules1234** [reply]
Thanks, I think I fixed it.

https://gist.github.com/16be6565412102abecbb8fdb20efadd7

**Reply by rbergen** [reply]
I can confirm that I no longer see the duplicate entries on the report page. Where we stand at the moment is that PrimeView is able to pull the results for 333 implementations across 69 languages out of your example JSON file. The counts for the project-generated JSON files are 373 implementations across 78 languages on the same date (December 28th).

**Reply by sonicrules1234** [reply]
Ok, it was missing the ones that didn't output a newline at the end of the run.  I think I've fixed it.

https://gist.github.com/a157bf979059421a7db5dfc1f67e3b59

**Reply by rbergen** [reply]
Well, both the implementation and language count do match, now.
